### PR TITLE
Ns1 config improvements

### PR DESCRIPTION
**Pull request by dblagojevic-daitan**
### What does this PR do?

Changes to configuration file to allow easier handling of zone records without specifying each record for the zone
Changes to allow reporting of more than one network usage statistic
Improved test coverage

### CODE REVIEW

**Review by ofek — APPROVED** [review]
🥇

File: ns1/datadog_checks/ns1/data/conf.yaml.example [hunk 1]
```diff
@@ -23,22 +23,35 @@ instances:
     #
     api_key: test_key_123
 
+    ## @param networks - list - optional
+    ## If present, usage stats will be queried and reported by network.
+    ## Each network's stats will be tagged with network name.
+    ## If omitted, usage stats will be reported only account wide.
+    ## If list of network id's is supplied only those networks will be queried.
+    ## If parameter is present without list of network id's,
+    ## all available networks will be queried.
```
Comment (kayayarai) [inline ns1/datadog_checks/ns1/data/conf.yaml.example:32]:
```suggestion
    ## If present, usage stats are queried and reported by the network.
    ## Each network's stats are tagged with network name.
    ## If omitted, usage stats are reported only account wide.
    ## If a list of network IDs is supplied, only those networks are queried.
    ## If the parameter is present without a list of network IDs,
    ## all available networks are queried.
```

File: ns1/datadog_checks/ns1/data/conf.yaml.example [hunk 2]
```diff
@@ -23,22 +23,35 @@ instances:
...
+    ## Each individual entry within the metric dictionary is optional,
     ## however at least one has to be present
-    ## i.e. you must specify at least one of following:
+    ## i.e. you must specify at least one of the following:
     ## qps, usage, account, ddi, pulsar, pulsar_by_app, pulsar_by_record
     #
     metrics:
     ## @param pulsar - boolean - optional
-    ## if parameter is present following will be repoted account-wide:
+    ## if parameter is present following will be reported account-wide:
```
Comment (kayayarai) [inline ns1/datadog_checks/ns1/data/conf.yaml.example:48]:
```suggestion
    ## if parameter is present, the following is reported account-wide:
```

File: ns1/datadog_checks/ns1/data/conf.yaml.example [hunk 3]
```diff
@@ -23,22 +23,35 @@ instances:
...
     ## number of decisions, number of undetermined decisions, route map hits and misses
     #
     #   pulsar:
 
     ## @param pulsar_by_record - list - optional
-    ## if parameter is present following will be repoted on per record basis:
+    ## if parameter is present, the following will be reported on per record basis:
```
Comment (kayayarai) [inline ns1/datadog_checks/ns1/data/conf.yaml.example:54]:
```suggestion
    ## if parameter is present, the following is reported on per record basis:
```

**Review by kayayarai — CHANGES_REQUESTED** [review]
Edits from docs

**Review by kayayarai — APPROVED** [review]

**Pull request merged**